### PR TITLE
Replace tslint with eslint

### PR DESCRIPTION
**Pull request by roggervalf**
tslint is deprecated as in https://code.visualstudio.com/api/advanced-topics/tslint-eslint-migration, this PR is trying to migrate to eslint and also set pretty-quick, so the code could be formatted before committing

### CODE REVIEW

**Comment by manast** [comment]
Is it possible to configure eslint so that it is compatible with current formatting? that would help a lot in order to replace tslint smoothly.

**Comment by roggervalf** [comment]
> Is it possible to configure eslint so that it is compatible with current formatting? that would help a lot in order to replace tslint smoothly.

sure, let me see

File: src/test/fixtures/delay.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
+'use strict';
+
 module.exports = function delay(ms) {
-  return new Promise(function(resolve) {
+  return new Promise(function (resolve) {
```
Comment (manast) [inline src/test/fixtures/delay.js:4]: there is a new space now, is this because of eslint or why this change?

Comment (roggervalf) [reply src/test/fixtures/delay.js]: yep, there was a missing rule in eslint

**Comment by manast** [comment]
excelent job!
